cob_twist_controller/src/inverse_jacobian_calculations/inverse_jacobian_calculation.cpp

```diff
@@ -30,6 +30,32 @@
 #include <Eigen/SVD>
 #include <ros/ros.h>
 
+
+
+/**
+ * Calculates the pseudoinverse of the Jacobian by using SVD technique.
+ * This allows to get information about singular values and evaluate them.
+ */
+Eigen::MatrixXd PInvBySVD::calculate(const Eigen::MatrixXd& jacobian) const
+{
+    Eigen::JacobiSVD<Eigen::MatrixXd> svd(jacobian, Eigen::ComputeThinU | Eigen::ComputeThinV);
+    double eps_truncation = DIV0_SAFE;  //prevent division by 0.0
+    Eigen::VectorXd singularValues = svd.singularValues();
+    Eigen::VectorXd singularValuesInv = Eigen::VectorXd::Zero(singularValues.rows());
+
+    // small change to ref: here quadratic damping due to Control of Redundant Robot Manipulators : R.V. Patel, 2005, Springer [Page 13-14]
+    for(uint32_t i = 0; i < singularValues.rows(); ++i)
+    {
+        double denominator = singularValues(i) * singularValues(i);
+        //singularValuesInv(i) = (denominator < eps_truncation) ? 0.0 : singularValues(i) / denominator;
+        singularValuesInv(i) = (singularValues(i) < eps_truncation) ? 0.0 : singularValues(i) / denominator;
```

Comment:
I think just calculation 1.0 / singularValues(i) would be enough to avoid unnecessary multiplication in line 49.